Sources/Sentry/Public/SentryOptions.h

```diff
@@ -28,15 +28,18 @@ NS_SWIFT_NAME(Options)
 @property (nullable, nonatomic, strong) SentryDsn *parsedDsn;
 
 /**
- * Turns debug mode on or off. If debug is enabled SDK will attempt to print out useful debugging
- * information if something goes wrong. Default is disabled.
+ * Turns debug mode on or off. If debug is enabled, the SDK will print verbose console logs; if it
+ * is disabled, it only prints info or higher unless @c diagnosticLevel is set to a different level.
+ * Default is disabled.
  */
 @property (nonatomic, assign) BOOL debug;
 
 /**
- * Minimum LogLevel to be used if debug is enabled. Default is debug.
+ * Minimum LogLevel to be used if debug is disabled. Default is debug.
  */
-@property (nonatomic, assign) SentryLevel diagnosticLevel;
+@property (nonatomic, assign) SentryLevel diagnosticLevel DEPRECATED_MSG_ATTRIBUTE(
+    "This property will be removed in a future version of the SDK. The default log level will be "
+    "info, or you can set `debug` to `YES` to enable debug logging.");
```

Comment:
m: Im not sure about deprecating this.

@bruno-garcia I believe  you started this. What do you think?

Reply:
This is available in other sdks and is very useful there. Isn't it here? Is it because levels are not well distributed?

Reply:
I believe level distribution is indeed a problem.

Reply:
In general we defined the sdk shouldn't emit any logs by default not to polute the users dev env.
That mindset is more of a terminal thing I imagine. We did this differently in Unity where the default level is warning and it's on by default (meaning Debug is true with diag level warning).
Users can turn off logging by setting Debug to false. Then. Not even error logs show up.

Reply:
Yeah I see where you all are coming from with this, I will revert this change to the API.

Regarding the overall theme of this PR though, I just found someone who was looking for some logs in a production app, and they mention that they do get them in their android app. Is it because they configured their android app differently, or does android do what this PR proposes (to output logs in nondebug mode, just at a higher level)? https://discord.com/channels/621778831602221064/816438007921377361/984379663952535582

Reply:
OK, it appears that Android does do the same thing: https://github.com/getsentry/sentry-java/blob/bf18c0d5b0dfc8c1df5a1a6cda33baf1102fbaaf/sentry/src/main/java/io/sentry/DiagnosticLogger.java#L38

So I can also revert that change in functionality here so they don't diverge. If there's nothing useful left afterwards I can just close this PR.